content/doc/pipeline/index.adoc

```diff
@@ -1,981 +1,437 @@
 
-```groovy
-archive includes: '*.jar', excludes: '*-sources.jar'
-```
+node {
+  git url: 'https://github.com/joe_user/simple-maven-project-with-tests.git'
+  def mvnHome = tool 'M3'
+  env.PATH = "${mvnHome}/bin:${env.PATH}"
+  sh 'mvn -B verify'
+}
 
-Archives allow you to maintain binaries from your build in Jenkins for easy access later. Unlike stash, archive keeps artifacts around after a pipeline execution is complete (where stash is temporary).
+* Properties of this variable are environment variables on the current node.
+* You can override certain environment variables and the overrides are seen by subsequent sh steps (or anything else that pays attention to environment variables).
+* You can run mvn without a fully-qualified path.
 
-Beyond stashing and archiving files, the following Pipeline elements also work with the file system (more details in full reference card):
+Setting a variable such as PATH in this way is only safe if you are using a single agent for this build. As an alternative, you can use the withEnv step to set a variable within a scope:
 
-```groovy
-pwd()
-dir(''){}
-writeFile file: 'target/results.txt', text: ''
-readFile 'target/results.txt'
-fileExists 'target/results.txt'
-```
+ node {
+   git url: 'https://github.com/jglick/simple-maven-project-with-tests.git'
+   withEnv(["PATH+MAVEN=${tool 'M3'}/bin"]) {
+     sh 'mvn -B verify'
+   }
+ }
 
-# Scaling your Pipeline
+Jenkins defines some environment variables by default:
 
-As you build more of your DevOps pipelines with Jenkins Pipeline, your needs will get more complex. The CloudBees Jenkins Platform helps scale Jenkins Pipeline for more complex uses.
+*Example:* env.BUILD_TAG can be used to get a tag like jenkins-projname-1 from Groovy code, or $BUILD_TAG can be used from a sh script.
+The Snippet Generator help for the withEnv step has additional detail on this topic.
 
-## Checkpoints
-One powerful aspect of the CloudBees extensions to Jenkins Pipeline is the checkpoint syntax. Checkpoints allow capturing the workspace state so it can be reused as a starting point for subsequent runs:
+=== Build Parameters
 
-```groovy
-checkpoint 'Functional Tests Complete'
-```
+If you configured your pipeline to accept parameters using the *Build with Parameters* option, those parameters are accessible as Groovy variables of the same name.
 
-Checkpoints are ideal to use after a longer portion of your pipeline has run, for example a robust functional test suite.
+=== Recording Test Results and Artifacts
 
-## Pipeline Templates
+If there are any test failures in a given build, you want Jenkins to record them, and then proceed, rather than stopping. If you want it saved, you must capture the JAR that you built. The following sample code for a node shows how (As previously seen in several examples from this guide, Maven is being used as a build tool):
```

Comment:
These sentences are jarring and confusing to me

> If there are any test failures in a given build, you want Jenkins to record them, and then proceed, rather than stopping.

What is being said here?

> If you want it saved, you must capture the JAR that you built.

Same here?

I assume both are trying to set up the context for why recording test results and persisting artifacts are both useful. but the Shatner-esque rhetorical questions fall flat IMO